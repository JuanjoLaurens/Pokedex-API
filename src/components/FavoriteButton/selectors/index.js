const selectors = {
  favorites: (state) => state.favorites.data,
};

export default selectors;
