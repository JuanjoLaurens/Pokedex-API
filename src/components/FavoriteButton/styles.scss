@import '../../scss/bootstrap';

.pd-favorite {
  transition: .2s transform;
  cursor: pointer;
  display: flex !important;
  justify-content: center;
  align-items: center;
  border-radius: 50% !important;
  border: none !important;
  height: 50px;
  width: 50px;
  background-color: $white !important;

  outline: none !important;

  &:hover {
    transform: scale(1.1);
  }
}
