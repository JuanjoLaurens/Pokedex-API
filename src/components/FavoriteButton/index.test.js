import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

import FavoriteButton from './index';

jest.mock('react-redux', () => ({
  useSelector: jest.fn((value) => value),
  useDispatch: jest.fn(() => () => {}),
}));

const makeProps = ({ src, name, index } = {}) => ({
  src: src || null,
  name: name || 'pikachu',
  index: index || 1,
});

const renderComponent = (props) => {
  return shallow(<FavoriteButton {...props} />);
};

describe('FavoriteButton', () => {
  it('renders correctly according to snapshot', () => {
    const wrapper = renderComponent(makeProps());

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
