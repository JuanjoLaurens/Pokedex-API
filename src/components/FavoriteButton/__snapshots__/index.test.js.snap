// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteButton renders correctly according to snapshot 1`] = `
<button
  className="pd-favorite shadow"
  onClick={[Function]}
  type="button"
  variant="primary"
>
  <span
    className="material-icons text-muted"
  >
    favorite_border
  </span>
</button>
`;
